analytical_engine/core/context/java_context_base.h

```diff
@@ -0,0 +1,355 @@
+
+      // 2. Create Message manager Java object
+      jobject messagesObject =
+          createFFIPointer(env, java_message_manager_name,
+                           url_class_loader_object_, messages_addr);
+      CHECK_NOTNULL(messagesObject);
+      mm_object_ = env->NewGlobalRef(messagesObject);
+
+      // 3. Create arguments array
+      {
+        jclass json_class = (jclass) load_class_with_class_loader(
+            env, url_class_loader_object_, JSON_CLASS_NAME);
+        if (env->ExceptionCheck()) {
+          env->ExceptionDescribe();
+          env->ExceptionClear();
+          VLOG(0) << "Exception in loading json class ";
+        }
+        CHECK_NOTNULL(json_class);
+        jmethodID parse_method = env->GetStaticMethodID(
+            json_class, "parseObject",
+            "(Ljava/lang/String;)Lcom/alibaba/fastjson/JSONObject;");
+        CHECK_NOTNULL(parse_method);
+        VLOG(2) << "User defined kw args: " << args_str;
+        jstring args_jstring = env->NewStringUTF(args_str.c_str());
+        jobject json_object =
+            env->CallStaticObjectMethod(json_class, parse_method, args_jstring);
+        CHECK_NOTNULL(json_object);
+
+        // 4. Invoke java method
+        env->CallVoidMethod(context_object_, init_method_id, fragment_object_,
+                            mm_object_, json_object);
+        if (env->ExceptionCheck()) {
+          env->ExceptionDescribe();
+          env->ExceptionClear();
+          VLOG(0) << "Exception in context Init";
+        }
+        VLOG(2) << "Successfully invokd ctx init method.";
+        // 5. to output the result, we need the c++ context held by java object.
+        jfieldID inner_ctx_address_field =
+            env->GetFieldID(context_class, "ffiContextAddress", "J");
+        CHECK_NOTNULL(inner_ctx_address_field);
+
+        inner_ctx_addr_ =
+            env->GetLongField(context_object_, inner_ctx_address_field);
+        CHECK_NE(inner_ctx_addr_, 0);
+        VLOG(2) << "Successfully obtained inner ctx address";
+      }
+    }
+  }
+
+ private:
+  // user library name should be absolute
+  std::string parse_params_and_setup_jvm_env(const std::string& params,
+                                             std::string& user_library_name,
+                                             std::string& user_class_path) {
+    boost::property_tree::ptree pt;
+    std::stringstream ss;
+    {
+      ss << params;
+      try {
+        boost::property_tree::read_json(ss, pt);
+      } catch (boost::property_tree::ptree_error& r) {
+        VLOG(0) << "parse json failed: " << params;
+      }
+    }
+
+    VLOG(2) << "Received json: " << params;
+    std::string frag_name = pt.get<std::string>("frag_name");
+    CHECK(!frag_name.empty());
+    VLOG(2) << "Parse frag name: " << frag_name;
+    graph_type_str_ = frag_name;
+    pt.erase("frag_name");
+
+    std::string app_class_name = pt.get<std::string>("app_class");
+    CHECK(!app_class_name.empty());
```

Comment:
No such `CHECK()`s there, instead, returns `bl::result<std::string>` as result if your method may fail to make sure the error can be propogated to client.

Reply:
It would be great if these errors can be returned back to client. But as far as I'm concerned, when ```worker->context.init()```, It doesn't provided any interface/mechanism for GS error stuff. 
Any idea for this?